test/albumController.test.js

```diff
@@ -0,0 +1,190 @@
+    const response = await request(app)
+      .get(`/api/v1/albums/${album._id}/listOfUsersWhoPurchasedThisAlbum`)
+      .set('Cookie', signedCookie); // Set the obtained cookie
+
+    // Check if the response status is OK and contains the expected properties
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('album');
+    expect(response.body).toHaveProperty('purchasingUsersCount');
+  });
+
+  it('should test the getFilteredAlbums endpoint - Success Case', async () => {
+    // Create 1 album in the database
+    const albumToCreate = {
+      albumName: 'Album 1',
+      artistName: 'Artist 1',
+      spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+    };
+
+    await Album.create(albumToCreate);
+
+    const response = await request(app).get(
+      '/api/v1/albums?limit=10&order=asc'
+    );
+
+    expect(response.status).toBe(StatusCodes.OK);
+    // Assert that the response body is an object with "albums" and "count" keys
+    expect(response.body).toHaveProperty('albums');
+    expect(response.body).toHaveProperty('count');
+    // Assert that "albums" is an array with the expected number of albums (based on the limit)
+    // In this case, since we have only 1 album in the database, the "albums" array should have 1 item
+    expect(response.body.albums).toHaveLength(1);
+
+    // Assert that the "count" field reflects the actual count of albums in the database
+    // In this case, it should be 1 because we created 1 album
+    expect(response.body.count).toBe(1);
+  });
+
+  it('should return a limited list of albums when using a limit parameter', async () => {
+    const albumsToCreate = [
+      {
+        albumName: 'Album 1',
+        artistName: 'Artist 1',
+        price: 9.99,
+        image: '/uploads/album1.jpg',
+        releaseDate: '2022-01-01T00:00:00.000Z',
+        category: 'rock',
+        spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+        averageRating: 4.5,
+        numOfReviews: 2,
+      },
+      {
+        albumName: 'Album 2',
+        artistName: 'Artist 2',
+        price: 8.99,
+        image: '/uploads/album2.jpg',
+        releaseDate: '2022-02-01T00:00:00.000Z',
+        category: 'pop',
+        spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+        averageRating: 3.7,
+        numOfReviews: 3,
+      },
+      {
+        albumName: 'Album 3',
+        artistName: 'Artist 3',
+        price: 7.99,
+        image: '/uploads/album3.jpg',
+        releaseDate: '2022-03-01T00:00:00.000Z',
+        category: 'rock',
+        spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+        averageRating: 4.2,
+        numOfReviews: 1,
+      },
+    ];
+
+    await Album.insertMany(albumsToCreate);
+
+    // Make a GET request to the /api/v1/albums endpoint with a limit parameter set to 2
+    const limit = 2;
+    const response = await request(app).get(`/api/v1/albums?limit=${limit}`);
```

Comment:
Same as above comment about the correct URL to use the filter endpoint that takes a bunch of query parameters:

```suggestion
    const response = await request(app).get(`/api/v1/albums/filter?limit=${limit}`);
```

Reply:
Thank you 👍